### DISCUSSION TITLE
Handling Setting Changes along with queued actions

### DISCUSSION THREAD

**Discussion by ragingdave** [spatie/laravel-settings#58] [Q&A]
I came across an oddity/edge case (maybe?) around using settings in queue jobs/mails/etc. Changing a setting doesn't update the queue workers as those are long lived and in tandem with the caching, makes it so that the value from settings doesn't change until at least horizon restarts itself.

One solution I thought of was to trigger a call (only queued based on https://github.com/laravel/framework/issues/28217) to `horizon:terminate` in the SettingsSaved event, which would theoretically short term fix the issue, but I feel like that's just a band aid, more than a fix.

What are the thoughts on how best to approach the issue?

**Comment by SebastiaanKloos** [comment]
@ragingdave Maybe it's a little late for an answer. But I was experiencing an issue with this also. I fixed it today by adding a function to the queue which runs before each job.

```php
Queue::before(function () {
    app(SiteSettings::class)->refresh();

    $this->bootMailSettings();
});
```
I added this to the boot method of my service provider. It refreshes the settings and sets up some mailer configuration in my case.

Hope this helps!

**Comment by judgej** [comment]
2024 and hit this problem also. Thanks for the tip @SebastiaanKloos
